May release notes (#3172)
adobe/react-spectrum: packages/dev/docs/pages/releases/2022-05-27.mdx

{/* Copyright 2022 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. */}

import {BlogPostLayout, Hero} from '@react-spectrum/docs';
export default BlogPostLayout;

---
description: React 18 is now officially supported! Other updates include a major version bump to our workflow icons package, support for emphasized Tabs, and various bug fixes to our existing components and hooks. Thank you to all who contributed to this release!
date: 2022-05-27
---

> # May 27, 2022 Release

Reviewer: ```suggestion
# May 31, 2022 Release
```

{/* Copyright 2022 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. */}

import {BlogPostLayout, Hero} from '@react-spectrum/docs';
export default BlogPostLayout;

---
description: React 18 is now officially supported! Other updates include a major version bump to our workflow icons package, support for emphasized Tabs, and various bug fixes to our existing components and hooks. Thank you to all who contributed to this release!
date: 2022-05-27
---

# May 27, 2022 Release

> Ever since React 18 was announced last July, compatibility has been front of mind for the team. We are extremely pleased to announce that our library now officially supports React 18! Special thanks to [@eps1lon](https://github.com/eps1lon) for their
> help with getting our tests working with React 18.

Reviewer: ```suggestion
This release includes compatibility with React 18! We have tested each component and hook to verify that it works as expected, while retaining support for React 17 and 16 as well. Special thanks to [@eps1lon](https://github.com/eps1lon) for their help with getting our tests working with React 18. Note that this release only includes basic compatibility. Strict mode support is still a work in progress, and will be supported in a future release.
```

> A major version bump to our workflow icons package is included in this release, so please check out the changes [here](#icons-update) and upgrade accordingly.

Reviewer: ```suggestion
A major version bump to our Spectrum workflow icons package is included in this release, so please check out the changes [here](#icons-update) and upgrade accordingly.
```

you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. */}

import {BlogPostLayout, Hero} from '@react-spectrum/docs';
export default BlogPostLayout;

---
description: React 18 is now officially supported! Other updates include a major version bump to our workflow icons, support for emphasized Tabs, and various bug fixes to our existing components and hooks. Thank you to all who contributed to this release!
date: 2022-05-27
---

# May 27, 2022 Release

Ever since React 18 was announced last July, compatibility has been front of mind for the team. We are extremely pleased to announce that React 18 is now officially supported with our library! Special thanks to [@eps1lon](https://github.com/eps1lon) for his

> help with getting our tests working with React 18.

Author: Open to feedback on wording here

Reviewer 2: I think it's good.

One thought, should the portion on Spectrum 18 and RSP library compatibility be flipped so it's more like “our library officially supports Spectrum 18.”

Reviewer 3: ```suggestion
Ever since React 18 was announced last July, compatibility has been front of mind for the team. We are extremely pleased to announce that React 18 is now officially supported by our library! Special thanks to [@eps1lon](https://github.com/eps1lon) for his
help with getting our tests working with React 18.
```

> A major version bump to our workflow icons is included in this release so be sure to upgrade accordingly. Other updates include added support for emphasized Tabs and various bug fixes to our existing components and hooks. Thank you to all who contributed to this release!

Reviewer 2: Do we need to explain what the icon change means, i.e. added, removed, and changed? Or do they need to read Rob's PR?

> A major version bump to our workflow icons is included in this release so be sure to upgrade accordingly. Other updates include added support for emphasized Tabs and various bug fixes to our existing components and hooks. Thank you to all who contributed to this release!

Reviewer 3: ```suggestion
A major version bump to our workflow icons is included in this release, so be sure to upgrade accordingly. Other updates include added support for emphasized Tabs and various bug fixes to our existing components and hooks. Thank you to all who contributed to this release!
```

- Upgrade workflow icons - [@snowystinger](https://github.com/snowystinger) - [PR](https://github.com/adobe/react-spectrum/pull/3019)

## Fixes
- Reduce the number of rerenders from focus hooks - [@csantos1113](https://github.com/csantos1113) - [PR](https://github.com/adobe/react-spectrum/pull/3045)
- Fix NumberParser mistakenly returning Arabic values in non-Arabic locales  - [@snowystinger](https://github.com/snowystinger) - [PR](https://github.com/adobe/react-spectrum/pull/3116)
- Add ActionMenu to mono-package - [@snowystinger](https://github.com/snowystinger) - [PR](https://github.com/adobe/react-spectrum/pull/3144)
- Distinguish Talkback virtual clicks from Windows FF touch event - [@LFDanLu](https://github.com/LFDanLu) - [PR](https://github.com/adobe/react-spectrum/pull/3136)
- Fixes missing 'props' from API sections of useFocus and useFocusWithin docs - [@ktabors](https://github.com/ktabors) - [PR](https://github.com/adobe/react-spectrum/pull/3161)
- Fix ButtonGroup visual flicker on resize check - [@LFDanLu](https://github.com/LFDanLu) - [PR](https://github.com/adobe/react-spectrum/pull/3160)

> - Fix React error for Tabs when every key is disabled - [@snowystinger](https://github.com/snowystinger) - [PR](https://github.com/adobe/react-spectrum/pull/3162)

Reviewer 4: technically this fixed something we broke in between releases, so it shouldn't be part of the release notes. if it'd been broken in the last release, then it would make sense here

## Under construction

Fixes to pre-released components are listed below. Please feel free to try them out, and report any issues you encounter.

- Fix edge cases in DatePicker with min/max dates and eras - [@devongovett](https://github.com/devongovett) - [PR](https://github.com/adobe/react-spectrum/pull/3149)

> - Make DatePicker button keyboard focusable - [@majornista](https://github.com/majornista) - [PR](https://github.com/adobe/react-spectrum/pull/3145)

Author: Do we wanna call these out?

Reviewer 2: Maybe just in the Slack annoucement?

Author: Maybe? There is precedent for announcing updates to pre-release components though so I'm not sure

Reviewer 5: usually I just put the pre-release packages that were updated and their docs.  I guess in this case it would be all of them because of the react 18 change

Author: right, I'm assuming we don't wanna list them all. The precedent I'm referring to is https://react-spectrum.adobe.com/releases/2022-03-21.html though where we announced specific pre-release component changes

you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. */}

import {BlogPostLayout, Hero} from '@react-spectrum/docs';
export default BlogPostLayout;

---
description: React 18 is now officially supported! Other updates include a major version bump to our workflow icons, support for emphasized Tabs, and various bug fixes to our existing components and hooks. Thank you to all who contributed to this release!
date: 2022-05-27
---

# May 27, 2022 Release

Ever since React 18 was announced last July, compatibility has been front of mind for the team. We are extremely pleased to announce that our library now officially supports React 18! Special thanks to [@eps1lon](https://github.com/eps1lon) for their
help with getting our tests working with React 18.

> A major version bump to our workflow icons is included in this release, so please check out the changes [here](#icons-update) and upgrade accordingly.

Reviewer 5: ```suggestion
A major version bump to our workflow icons package is included in this release, so please check out the changes [here](#icons-update) and upgrade accordingly.
```

Ever since React 18 was announced last July, compatibility has been front of mind for the team. We are extremely pleased to announce that our library now officially supports React 18! Special thanks to [@eps1lon](https://github.com/eps1lon) for their
help with getting our tests working with React 18.

> A major version bump to our workflow icons is included in this release, so be sure to upgrade accordingly. The changes in this version update are as follows:

Reviewer 4: I feel like this should be a section down below instead of being in the overview. Overview should say something like
```
A major version bump to our workflow icons is included in this release because our icons team released a major version, so be sure to upgrade according to the "Name of section" below.
```

> 3 icons were renamed to fix typo in name:

Reviewer 4: ```suggestion
3 icons were renamed to fix a typo in their names:
```
